tests/Feature/WebRouteTest.php

```diff
@@ -162,4 +162,25 @@ public function test_store_import_status()
         $response->assertSuccessful();
         $response->assertViewIs('importStatus');
     }
+
+    /**
+     * Test error response handling
+     *
+     * @return void
+     */
+    public function test_error_response_handling()
+    {
+        $redirect = $this
+            ->get(
+                route('results', ['ids' => 'Q1']),
+                ['X-Mismatch-Results-Error' => 'true']
+            );
+
+        $redirect->assertRedirect(route('home'));
+        $response = $this->get($redirect->headers->get('Location'));
+        $response->assertInertia(function (Assert $page) {
+                $page->component('Home')
+                    ->where('flash.errors', [ 'unexpected' => 'Unexpected error']);
+        });
+    }
```

Comment:
You were right, I love this! I'm really impressed with all the testing helpers Laravel and inertia provide us with :100:

Reply:
added a comment to make more explicit that we first get a redirect response, which we then follow to the URL from the 'Location' header.